system/jlib/jprop.cpp

```diff
@@ -402,6 +402,40 @@ IProperties *cloneProperties(const IProperties * source, bool nocase)
     return clone.getClear();
 }
 
+//This works on arrays of string of the form x=y and x: y
+void extractHeaders(IProperties * target, const StringArray & httpHeaders, char separator)
+{
+    StringBuffer key;
+    ForEachItemIn(currentHeaderIndex, httpHeaders)
+    {
+        const char* httpHeader = httpHeaders.item(currentHeaderIndex);
+        if(isEmptyString(httpHeader))
+            continue;
+
+        const char* delineator = strchr(httpHeader, separator);
+        if ((delineator == nullptr) || (delineator == httpHeader))
+            continue;
+
+        const char * value = delineator + 1;
+        while (isspace(*value))
+            value++;
+
+        if (*value)
+        {
+            key.clear().append(delineator - httpHeader, httpHeader);
```

Comment:
Not necessary for our use cases, but would be pretty easy to remove trailing whitespace before dlineator if you wanted to be flexible about supporting "x = y".  Or would this be a case of being strict is actually better?

Reply:
I think revisit if there is a use-case it would actually be useful for.